src/main/java/gregtech/common/items/behaviors/ItemMagnetBehavior.java

```diff
@@ -40,15 +40,38 @@ public ItemMagnetBehavior(int range, float speed) {
     @Override
     public ActionResult<ItemStack> onItemRightClick(World world, @Nonnull EntityPlayer player, EnumHand hand) {
         if (!player.world.isRemote && player.isSneaking()) {
-            this.isActive = !this.isActive;
-            player.sendMessage(new TextComponentTranslation(isActive ? "behavior.item_magnet.enabled" : "behavior.item_magnet.disabled"));
+            player.sendStatusMessage(new TextComponentTranslation(toggleActive(player.getHeldItem(hand)) ? "behavior.item_magnet.enabled" : "behavior.item_magnet.disabled"), true);
         }
         return ActionResult.newResult(EnumActionResult.PASS, player.getHeldItem(hand));
     }
 
+    private boolean isActive(ItemStack stack) {
+        if (stack == ItemStack.EMPTY || stack == null) {
```

Comment:
Normally you don't need a stack null check. If a stack is null something is really wrong.
Rather use `stack.isEmpty()` then `stack == ItemStack.EMPTY`